kernel/src/transaction.rs

```diff
@@ -138,15 +141,14 @@ impl Transaction {
             .map(|txn| txn.into_engine_data(get_log_txn_schema().clone(), engine));
 
         // step one: construct the iterator of commit info + file actions we want to commit
-        let engine_commit_info = self
-            .commit_info
-            .as_ref()
-            .ok_or_else(|| Error::MissingCommitInfo)?;
+        if self.engine_commit_info.is_empty() {
+            return Err(Error::MissingCommitInfo);
+        }
```

Comment:
I think we don't have to require it now so we can remove this error case

Reply:
if user doesn't call this API we should just write out our own commit info with no engine "stuff" in it